### PR TITLE
Refs [TASK] [36706] Gọi API tìm kiếm hình ảnh

### PR DESCRIPTION
**Pull request by luongth-2520**
## Related Tickets
- [#Task 36706: Gọi API tìm kiếm hình ảnh](https://edu-redmine.sun-asterisk.vn/issues/36706)
## WHAT
- Gọi API tìm kiếm hình ảnh
## Evidence (Screenshot or Video)
Screen Shot 2021-05-28 at 15 30 25

## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://github.com/framgia/coding-standards/tree/master/eng/android |<li>- [ ] yes</li>|<li>- [ ] yes</li>
Conventions | Kotlin coding conventions | https://kotlinlang.org/docs/coding-conventions.html |<li>- [ ] yes</li>|<li>- [ ] yes</li>
Redmine | Sun* Redmine working process  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md |<li>- [ ] yes</li>|<li>- [ ] yes</li>

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/sun/unsplash_02/data/source/ImageDataSource.kt [hunk 1]
```diff
@@ -19,5 +19,9 @@ interface ImageDataSource {
             collectionId: String,
             page: Int
         )
+        fun searchImage(
+            listener: OnFetchDataJsonListener<MutableList<Image?>>,
+            query: String, page: Int
+        )
```
Comment (khoaqh-2470) [inline app/src/main/java/com/sun/unsplash_02/data/source/ImageDataSource.kt]:
```suggestion
        fun searchImage(
            listener: OnFetchDataJsonListener<MutableList<Image?>>,
            query: String,
            page: Int
        )
```

**Review by khoaqh-2470 — APPROVED** [review]

File: app/src/main/java/com/sun/unsplash_02/screen/search/SearchFragment.kt [hunk 2]
```diff
@@ -1,22 +1,121 @@
...
+import androidx.recyclerview.widget.RecyclerView
 import com.sun.unsplash_02.R
 import com.sun.unsplash_02.base.BaseFragment
+import com.sun.unsplash_02.data.model.Image
+import com.sun.unsplash_02.data.source.ImageRepository
+import com.sun.unsplash_02.screen.home.adapter.PhotoAdapter
+import com.sun.unsplash_02.screen.main.MainActivity
+import com.sun.unsplash_02.utils.hideKeyboard
+import com.sun.unsplash_02.utils.showKeyboard
+import kotlinx.android.synthetic.main.fragment_home.*
```
Comment (khoaqh-2470) [inline app/src/main/java/com/sun/unsplash_02/screen/search/SearchFragment.kt]: remove line

File: app/src/main/java/com/sun/unsplash_02/screen/search/SearchFragment.kt [hunk 3]
```diff
@@ -1,22 +1,121 @@
...
 
-class SearchFragment : BaseFragment() {
+class SearchFragment : BaseFragment(), SearchContract.View, PhotoAdapter.ItemClickListener {
 
-    override fun getLayoutResourceId() = R.layout.fragment_search
+    private lateinit var searchPresenter: SearchPresenter
+    private var photoAdapter: PhotoAdapter? = null
+    private var isLoading = false
+
+    override fun getLayoutResourceId(): Int = R.layout.fragment_search
```
Comment (khoaqh-2470) [inline app/src/main/java/com/sun/unsplash_02/screen/search/SearchFragment.kt]:
```suggestion
    override fun getLayoutResourceId() = R.layout.fragment_search
```

**Review by khoaqh-2470 — APPROVED** [review]

File: app/src/main/java/com/sun/unsplash_02/screen/search/SearchPresenter.kt [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    SearchContract.Presenter {
+
+    private var view: SearchContract.View? = null
+
+    override fun searchImage(query: String) {
+        view?.showLoading()
+        imageRepository.searchImage(object : OnFetchDataJsonListener<MutableList<Image?>> {
+            override fun onSuccess(data: MutableList<Image?>) {
+                view?.hideLoading()
+                view?.onImageLoaded(data)
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/sun/unsplash_02/screen/search/SearchPresenter.kt]: scope function

File: app/src/main/java/com/sun/unsplash_02/screen/search/SearchPresenter.kt [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+            }
+
+            override fun onError(exception: Exception?) {
+                view?.hideLoading()
+                view?.onError(exception)
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/sun/unsplash_02/screen/search/SearchPresenter.kt]: scope function

**Review by tuanlvt-1117 — APPROVED** [review]

**Pull request merged**